apps/src/music/blockly/MusicBlocklyWorkspace.js

```diff
@@ -417,70 +389,25 @@ export default class MusicBlocklyWorkspace {
     return 'musicLabSavedCode' + getBlockMode();
   }
 
-  // Loads sources using the project manager.  Falls back to the provided default sources.
-  async loadSources(startSources) {
-    const projectResponse = await this.projectManager.load();
-    if (!projectResponse.ok) {
-      if (projectResponse.status === 404) {
-        // This is expected if the user has never saved before.
-        this.setStartSources(startSources);
-      }
-
-      // TODO: Error handling
-      return;
-    }
-
-    const {source, channel} = await projectResponse.json();
-    this.channel = channel;
-    if (source && source.source) {
-      const existingCodeJson = JSON.parse(source.source);
+  // Either initialize the workspace with the given loadedSources,
+  // or fall back to start sources.
+  loadSources(startSources, loadedSources) {
+    if (loadedSources && loadedSources.source) {
+      const existingCodeJson = JSON.parse(loadedSources.source);
```

Comment:
good question. The main difference here is when we load start sources we call save, whereas for loading existing sources we do not. However, since we have the change checking going anyway, it would probably be fine to call save every time, it just might get ignored for existing sources.

Reply:
Ah true, but yeah, should be fine to call save in either case. Also I'm thinking we may pull loading starter code into ProjectManager eventually? so this may go away regardless

Reply:
refactored!